src/hdmf/build/map.py

```diff
@@ -1202,12 +1203,16 @@ def construct(self, **kwargs):
                 val = const_args[argname]
             else:
                 continue
+            # reduce single entry arrays to scalars
+            if isinstance(val, (Dataset, np.ndarray)):
+                if val.shape == (1,):
+                    val = val[0]
```

Comment:
I'm not 100% sure whether just checking for single-length shape is sufficient. I think you may need to look at the docval constructor or the spec to see if an array is expected or if a scalar is fine.

Reply:
There are places in pynwb where an array is expected and a scalar is not, e.g.:
https://github.com/NeurodataWithoutBorders/pynwb/blob/50f934c9e7dd8e9dccada38500c3790b6aca3dc1/src/pynwb/file.py#L201

`'array_data`' allows only array-like objects. You would have to change that to allow the scalar types as well. 
https://github.com/hdmf-dev/hdmf/blob/8ddc8574123853215a4d6c43d27bd2aec908017d/src/hdmf/utils.py#L13

Reply:
@oruebel @rly Thanks both. I'll come up with some docval magic.

Reply:
@oruebel @rly I found a different approach which does not require me to understand docval :duck:.

Reply:
Ditch that, this is BS.